lib: posix: fix error code returns

nniranjhana opened the pull request:
Added or fixed appropriate returns of error
codes wherever possible in pthread.c file.

Part of fix for #9993

Signed-off-by: Niranjhana N <niranjhana.n@intel.com>

Commit:
lib: posix: add error code return

Added return of ESRCH error code in
pthread_getschedparam() when the
specified thread could not be found.

Signed-off-by: Niranjhana N <niranjhana.n@intel.com>

nniranjhana commented:
Will add missed error codes for whichever possible in pthread.c and update here.

andyross reviewed (approved):
Sure, though note this doesn't actually protect against garbage thread IDs, just nulls and reuse of stale threads.

My personal intuition says we want to **not** include tests like this.  Applications that need robustness vs. programming error will be using USERSPACE anyway, and there we can rely on the much more elaborate checking (though right now that's a fatal error, but eventually it won't be)

nniranjhana commented:
@andyross Makes sense, well then I guess this fix will do temporarily. Also, how would we protect against garbage thread IDs? Would that include checking against each live thread at the time? I saw that in `pthread_cancel` too, we do only this same limited check

Commit:
lib: posix: fix return error code

Added EAGAIN error code in pthread_create()
with fixing the EINVAL return as it is
limited to attribute checking.

Signed-off-by: Niranjhana N <niranjhana.n@intel.com>

Commit:
lib: posix: remove return error code

ENOTSUP is not being used correctly in
pthread_attr_setschedparam(), hence
replaced its check for EINVAL instead.

Signed-off-by: Niranjhana N <niranjhana.n@intel.com>
